ruby/gilded_rose.rb

```diff
@@ -65,4 +24,128 @@ def initialize(name, sell_in, quality)
   def to_s()
     "#{@name}, #{@sell_in}, #{@quality}"
   end
+end
+
+class ItemUpdaterFactory
+  def self.create_updater_for(item)
+    { "Sulfuras, Hand of Ragnaros" => LegendaryItemUpdater.new(item),
+      "Aged Brie" => AgedBrieItemUpdater.new(item),
+      "Backstage passes to a TAFKAL80ETC concert" =>
+        BackstagePassesItemUpdater.new(item),
+      "Conjured" => ConjuredItemUpdater.new(item)
+    }.fetch(lookup_name(item), ItemUpdater.new(item))

```

Comment:
Love a good map ^_^

Reply:
Consider making the `Conjured` thing a separate routine to try when the name match fails. It could be a big confusing to read when certain strings in the map are name matches and another is actually a match on a substring.

Reply:
@dhartunian I was definitely thinking of you as I mapped this map